primefaces-showcase/src/main/java/org/primefaces/showcase/view/data/tree/BasicView.java

```diff
@@ -69,4 +71,13 @@ public void init() {
     public TreeNode getRoot() {
         return root;
     }
+
+    public boolean customFilter(Object data, Object filter, Locale locale) {
+        String filterText = (filter == null) ? null : filter.toString().trim().toLowerCase(locale);
+        if (LangUtils.isBlank(filterText) || data == null) {
```

Comment:
Just a minor performance comment you can probably
Check data == null up front and bail out before checking filter text etc

Reply:
Okay. But what do you think about my previous remark? Should we pass the data or the tree node?

Reply:
Hmm maybe the whole tree node for more flexibility?

Reply:
I was thinking the same.. will change it